app/ui/warehouse_schedule.py

```diff
@@ -0,0 +1,131 @@
+import streamlit as st
+from modules import add_custom_modules
+import connection
+import session
+import config
+from table import build_table, Actions
+from crud.base import BaseOpsCenterModel
+from typing import ClassVar
+from reports_heatmap import heatmap
+from filters import BaseFilter
+
+# Load custom OpsCenter python modules
+if not add_custom_modules():
+    st.warning("Unable to load OpsCenter modules.")
+
+
+import warehouses  # noqa E402
+
+sess = session.reports().get_report()
+
+
+class WarehouseSummary(BaseOpsCenterModel):
+    col_widths: ClassVar[dict] = {
+        "warehouse": ("Warehouse", 2),
+        "size": ("Size", 1),
+        "autoscale": ("Autoscale", 0.5),
+        "spend": ("Spend", 1),
+        "utilization": ("Utilization", 1),
+        "enabled": ("Schedule Enabled", 0.5),
+    }
+    warehouse: str
+    size: str
+    autoscale: bool
+    spend: str
+    utilization: str
+    enabled: bool
+
+
+def display():
+    st.title("Manage Warehouses")
+    filter_container = st.expander("Filters", expanded=True)
+    with filter_container:
+        s1, s2 = st.columns(2)
+
+        with s1:
+            range = st.date_input(
+                "Date Range", value=[sess.get_report_start(), sess.get_report_end()]
+            )
+        with s2:
+            cols = st.columns([1, 1, 1, 1])
+            cols[0].button(
+                "7", on_click=sess.set_date_range_days, args=[7], key="last7"
+            )
+            cols[1].button(
+                "30", on_click=sess.set_date_range_days, args=[30], key="last30"
+            )
+            cols[2].button(
+                "90", on_click=sess.set_date_range_days, args=[90], key="last90"
+            )
+            cols[3].button(
+                "365", on_click=sess.set_date_range_days, args=[365], key="last365"
+            )
+
+        credit_cost = config.get_compute_credit_cost()
+        sql = f"""
+        with tmp as procedure()
```

Comment:
This is pretty hideous, but I see why you're doing it this way. I also had no idea this syntax existed.

I'm worried about this being dubious to test/maintain.

Reply:
what is hideous? Not sure I follow

Reply:
My current line of thinking is that we want to stop from introducing "heavy" logic into OpsCenter code because Streamlit is kind of a poison pill when it comes to dev-friendly testing.

I shouldn't have used the word "hideous" -- I meant it as "very complex" and that complexity would be something I'd like to be able to run some kind of test case over. However, now that I'm reading this again: this would be pretty gross to try to write a test to cover without putting together some fixture for warehouse sessions.

The other thing unclear to me was why you used an inline procedure and not a scripting block. Is that an API limitation?

Reply:
ive removed the inline procedure in favour of a  scripting block